### ISSUE TITLE
Timeshift Should Automatically Delete Oldest Snapshots When Disk Space is Low

### ISSUE THREAD

**Issue by hirntot** [linuxmint/timeshift#329]
after my [bug report](https://github.com/linuxmint/timeshift/issues/221) was not as successful as it could be, I would like to add a related feature request:

**Is your feature request related to a problem? Please describe.**

I've been frustrated a few times already when my system disk becomes critically low on space due to accumulated Timeshift snapshots. This has led to my system becoming unresponsive or unbootable a few times already, causing significant disruptions. For instance, I once had to deal with a full disk issue that left me unable to access my system or manage my snapshots effectively. Even deleting snapshots via command line was not possible (see [other issue](https://github.com/linuxmint/timeshift/issues/330)).

**Describe the solution you'd like**

I would like Timeshift to automatically delete the oldest snapshots when it detects that the disk space is critically low. Having an automatic cleanup mechanism would greatly enhance the reliability of Timeshift on systems with limited disk space. It would prevent scenarios where users are locked out of their systems due to full disks.

**My ideal solution includes:**

- **Automatic Deletion:** Timeshift should detect when the disk space is critically low and automatically delete the oldest snapshots to free up space.
- **Minimum Free Space:** Ensure that there is always a minimum amount of free space available on the system disk (e.g., 1GB) to prevent system crashes or the inability to create new snapshots.
- **User Notification:** Notify the user when automatic deletion has occurred and provide information about the space freed and any remaining issues.

**Describe alternatives you've considered**

As an alternative, I've considered manually managing snapshots and regularly monitoring disk space. However, this is cumbersome and prone to errors, especially when dealing with unexpected space constraints.

---
thanks for the great tool anyway!!

**Comment by Infraviolet200** [comment]
Very oldest is surely the last one you'd want to delete? Very oldest s probably the most important one, the one you made on a fresh system once you'd installed all your programs but before you did anything that could corrupt it and cause you to need to make a restoration.

**Comment by hirntot** [comment]
Well, the Auto-deletion could be activated only for Auto-created snapshots..

**Comment by aral-matrix** [comment]
Seconded, I just had to do support for a friend who is on Mint as a less experienced user, and timeshift broke his system by filling up the root partition to the brim (34GB timeshift files) on default settings.

I did not even know this tool existed until we did a check on what root level folders occupied how much disk space.

Please modify timeshift default settings to NEVER EVER EVER EVER EVER EVER fill the target disk beyond 5 GB free disk space, and make that setting configurable for larger (or lesser, at your own risk) margins!

**Comment by GoldenGlimpse** [comment]
Yes please! I messed up my system and had no clue the external disk was full because of Timeshift, it literally has 0 free bytes left.
It stopped making backups 2 weeks ago. So instead of being able to reverse 1 or 2 days back, i had to restore it 2 weeks back.

Maybe... just maybe... at least a simple warning from Timeshift? Like, "hey... disk is almost full, please fix it."

Thanks.

**Comment by ygerlach** [comment]
currently timeshift should already auto delete snapshots when they fall "out of range". For example, if you set it to only have "5 Daily" (and have your computer on each day) the oldest snapshot should be at most 5 Days old. Snapshots are not deleted when they have a comment or are manually created. See: #463

**Comment by aral-matrix** [comment]
Seeing as it has been over a year without any reaction from the devs on this issue, I think it's time to petition mint to kick timeshift out of the default installation, and replace it with a tool that is actually supported.

**Comment by hirntot** [comment]
@aral-matrix is there a better or even comarable tool though?

**Comment by aral-matrix** [comment]
> [@aral-matrix](https://github.com/aral-matrix) is there a better or even comarable tool though?

Don't know don't care, my emphasis was to not install it *by default* - make it an opt-in (or a forced user choice) during installation, with a disclaimer that when used, it can render the computer unusable.

**Comment by hirntot** [comment]
I opened this issue a while ago and, as a workaround for my own systems - and for my customers - I wrote a small tool that deletes the oldest Timeshift snapshots when space gets critical, not touching timeshift itself.

Since this topic still isn’t fully solved upstream, I’ve now published it on GitHub in case it helps others:

https://github.com/hirntot/timeshift-spaceshift
